File: GroceryStoreSimulatorWebAPI/Models/Order.cs
using Dapper.Contrib.Extensions;

namespace GroceryStoreSimulatorWebAPI.Models
{
    [Table("tOrder")]
    public class Order
    {
        [Key]
        public int OrderID { get; set; }
        public int LoyaltyId { get; set; }
        public DateTime DateTimeCreated { get; set; }
        public int StoreId { get; set; }
        public int OrderStatusId { get; set; }
        public double DeliveryCharge { get; set; }
        public string? Notes { get; set; }
        public string? DeliveryAddress { get; set; }
        public DateTime? DateTimeDelivered { get; set; }
        public bool? AllOrNone { get; set; }
    }
}

File: FrontEnd/GroceryStoreSimulatorAngularApp/grocery-store-simulator/src/app/orders/order.ts
export interface Order{

    orderID: number
    LoyaltyId: number
    dateTimeCreated: Date
    StoreId: number
    OrderStatusId: number
    DeliveryCharge: number
    Notes: string,
    deliveryAddress: string
    dateTimeDelivered: Date
    AllOrNone: string
    OrderDetails: OrderDetails[]
    Store: Store;
}

export interface Store{
    storeId: number;
    store : string
    storeNumber: string;
}

export interface OrderDetails{

    OrderDetailId: number
    OrderId: number
    ProductId: number
    Quantity: number
    TotalAmountChargedToCustomer: number
    Comment: string
    CouponDetailId: number
    UnavailableWhenOrderWasFilled: boolean

}